src/state/actions/workspace.js

```diff
@@ -68,3 +73,40 @@ export function toggleWorkspaceExposeMode() {
     type: ActionTypes.TOGGLE_WORKSPACE_EXPOSE_MODE,
   };
 }
+
+/**
+ * importWorkspace - action creator
+ */
+export function importWorkspace(stateExport) {
```

Comment:
Are there tests for this action? Also, do we want to try to refactor this to avoid multiple dispatches before merging (e.g. as in #2121)? I wonder if that'd help address some of @aeschylus's concerns too.

Reply:
Added one.

Do you suggest to bundle the whole import process into one single dispatch or only the three dispatches that are done when replacing or creating a new window?

Reply:
Yeah, I was wondering if it could be done in a single dispatch, and have the reducers pick up the elements or clear state as needed.